### PR TITLE
Issue 11978 remove .git of template

### PR DESCRIPTION
**Pull request by bonespiked**
Added removal of .git subdirectory on creation w --template-url of a git repo.
ran lint, pretty, tests - I successfully ran locally, but wouldn't mind additional eyes/checks as it's my first serverless PR.

Closes: #11978

### CODE REVIEW

File: lib/utils/download-template-from-repo.js [hunk 1]
```diff
@@ -291,6 +291,7 @@ async function downloadTemplateFromRepo(inputUrl, templateName, downloadPath) {
 
     if (isPlainGitURL(inputUrl)) {
       return spawn('git', ['clone', inputUrl, downloadServicePath]).then(() => {
+        fse.removeSync(path.join(dirName, '/.git'));
```
Comment (medikoo) [inline lib/utils/download-template-from-repo.js:294]: Let's in first PR refactor `downloadTemplateFromRepo` function into async/await, and then introduce this change, but using _async_ fs interface

**Review by medikoo — CHANGES_REQUESTED** [review]
@bonespiked great thanks for that update, yet it'll be nice if we first upgrade this part into async/await (see my comment)

Otherwise, I'm worried we may introducing a breaking change here. I suspect that there can be users for which leaving `.git` folder is desired (e.g. imagine some company instructs to start their projects with this command).

Still, I think that we can give it a go, but if we receive a report that it appears to be breaking will have to revert the change and rethink approach

**Comment by bonespiked** [comment]
> @bonespiked great thanks for that updae, yet it'll be nice if we first upgrade this part into async/await (see my comment)
> 
> Otherwise, I'm worried we may introducing a breaking change here. I suspect that there can be users for which leaving `.git` folder is desired (e.g. imagine some company instructs to start their projects with this command).
> 
> Still, I think that we can give it a go, but if we receive a report that it appears to be breaking will have to revert the change and rethink approach

I can't see a valid use case where telling someone to start a new project from a template - then updates for their project being applied to the template itself being a desired behavior.  If you wanted that, you'd just clone the project repo.  My example _is_ telling people to start a project with this command, and then their project-specific updates are updating the template repo.. :) That said - I hear you on listening for feedback.

File: lib/utils/download-template-from-repo.js (hunk 1 above)
Comment (bonespiked) [reply lib/utils/download-template-from-repo.js:294]: will do.  Thanks for note.

**Pull request closed without merging**